### DISCUSSION TITLE
Calling an endpoint from another endpoint

### DISCUSSION THREAD

**Discussion by lukehutch** [serverpod/serverpod#1276] [General]
Some of the code in the endpoints I want to reuse by calling from other endpoints. This works fine if the two endpoints are in the same class, but I can't see how to call an endpoint in one class from an endpoint in another class. Is this possible? `session` does not seem to have all the endpoints added, like `client` does.

If not, can the endpoints all be added to `session`, the way that modules are added currently on the server, and the way that endpoints are added to `client`?

**Comment by vlidholt** [comment]
Another solution can be to add the shared code in a separate class (either as static methods or in a singleton class). You can call the shared methods from multiple endpoints.

I think your idea still makes sense, there should be references to the other endpoints somewhere.

**Comment by odrya12** [comment]
I used a direct call to another endpoint in my project and passed the session directly. It worked for me.

**Reply by lukehutch** [reply]
But how? Endpoint methods are not static. You need a reference to the instance of the endpoint class. These are supposed to be instantiated once, by the server. Where are the instances stored? In the client, they are stored in the client itself. But in the server, there is no client object.

**Reply by odrya12** [reply]
Oh, in my case, I've created a static function that both endpoints use.

**Comment by Sandyhi5** [comment]
I also need to access several methods of one endpoint from other endpoint(s). Why is it not possible to make them accessible from the .db object just as many other native methods like find(), findFirstRow(), toString(), etc. are visible between endpoints?
It is highly desirable

**Reply by lukehutch** [reply]
It wouldn't be part of the `db` object, the endpoints should all be added to `session.endpoint` or similar. But yes. I have needed this many times.

**Reply by Sandyhi5** [reply]
The reason I suggested .db was that the methods that I like to create are doing extra on standard CRUD methods that are associated with the db. 
Meanwhile I have now split such methods into a separate shared.dart within endpoints dir (no endpoint suffix) in order to keep going (as per other comment in this thread)

**Reply by lukehutch** [reply]
You have to put CRUD methods in an endpoint (those methods do call the database via `db`), and you really need to do that anyway because you should be authenticating the user for almost any request that hits the DB.

The issue here is a completely separate issue: how to call one endpoint from another endpoint, without creating a static or global function.

**Reply by Sandyhi5** [reply]
Yeah, so I for now I am good
Will watch this thread for future progress
